eZ/Publish/API/Repository/Tests/UserServiceTest.php

```diff
@@ -1879,4 +1881,52 @@ private function createHash($login, $password, $type)
         /* PASSWORD_HASH_MD5_PASSWORD (1) */
         return md5($password);
     }
+
+    /**
+     * Test for the createUser() method.
+     *
+     * @see \eZ\Publish\API\Repository\UserService::createUser()
+     * @expectedException \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function testCreateUserInvalidPasswordHashTypeThrowsException()
+    {
+        $repository = $this->getRepository();
+        $signalSlotUserService = $repository->getUserService();
+
+        $signalSlotUserServiceReflection = new ReflectionClass($signalSlotUserService);
+        $userServiceProperty = $signalSlotUserServiceReflection->getProperty('service');
+        $userServiceProperty->setAccessible(true);
+        $userService = $userServiceProperty->getValue($signalSlotUserService);
+
+        $userServiceReflection = new ReflectionClass($userService);
+        $settingsProperty = $userServiceReflection->getProperty('settings');
+        $settingsProperty->setAccessible(true);
+
+        $defaultUserServiceSettings = $settingsProperty->getValue($userService);
+
+        /* BEGIN: Use Case */
+        $settingsProperty->setValue(
+            $userService,
+            [
+                'hashType' => 42424242, // Non-existing hash type
+            ] + $settingsProperty->getValue($userService)
+        );
+
+        try {
+            $this->createUserVersion1();
+        } catch (InvalidArgumentException $e) {
+            $this->assertEquals("Argument 'type' is invalid: Password hash type '42424242' is not recognized", $e->getMessage());
```

Comment:
Move this to the annotation `@expectedExceptionMessage`, keep throw below of course.

*Side: in 6.7 we still use phpunit 4.8, so I don't mention using `$this->expectException...` methods as they're not available in this version yet.*